packages/hdwallet-native/src/bitcoin.ts

```diff
@@ -215,10 +215,10 @@ export function MixinNativeBTCWallet<TBase extends core.Constructor<NativeHDWall
       return true;
     }
 
-    buildInput(coin: core.Coin, input: core.BTCSignTxInputNative): InputData | null {
-      return this.needsMnemonic(!!this.#seed, () => {
+    async buildInput(coin: core.Coin, input: core.BTCSignTxInputNative): Promise<InputData | null> {
```

Comment:
This is the core reason things weren't like this before -- because `buildInput()` was synchronous and technically part of the public interface. That had trickle-down effects that prevented `util.getKeyPair()` from being async and from there propagated out.

Now that I'm more comfortable that I have a full picture of the the overall library and its use cases, I'm confident that this method should be marked `protected` if it weren't for the fact that we're unable to do so because the mixin strategy we use is build on exporting class expressions, which doesn't support `protected` or `private`. Therefore, I don't think this counts as a modification to the public API, so we can change it without breaking things.